modules/grpc/bigquery/bigquery-worker.cpp

```diff
@@ -22,33 +22,63 @@
  */
 
 #include "bigquery-worker.hpp"
+#include "bigquery-dest.hpp"
 
 #include "compat/cpp-start.h"
 #include "logthrdest/logthrdestdrv.h"
+#include "scratch-buffers.h"
+#include "logmsg/type-hinting.h"
 #include "compat/cpp-end.h"
 
+#include <string>
+#include <sstream>
+#include <chrono>
+
+#include <grpc/grpc.h>
+#include <grpcpp/client_context.h>
+#include <grpcpp/create_channel.h>
+#include <grpcpp/security/credentials.h>
+#include <google/protobuf/compiler/importer.h>
+
 using syslog_ng::bigquery::DestinationWorker;
+using syslog_ng::bigquery::DestinationDriver;
+using google::protobuf::FieldDescriptor;
 
 struct _BigQueryDestWorker
 {
   LogThreadedDestWorker super;
   DestinationWorker *cpp;
 };
 
-
 DestinationWorker::DestinationWorker(BigQueryDestWorker *s) : super(s)
 {
+  DestinationDriver *owner = this->get_owner();
 
+  std::stringstream table_name;
+  table_name << "projects/" << owner->get_project()
+             << "/datasets/" << owner->get_dataset()
+             << "/tables/" << owner->get_table();
+  this->table = table_name.str();
 }
 
 DestinationWorker::~DestinationWorker()
 {
-
 }
 
 bool
 DestinationWorker::init()
 {
+  DestinationDriver *owner = this->get_owner();
+
+  this->channel = grpc::CreateChannel(owner->get_url(), grpc::GoogleDefaultCredentials());
```

Comment:
We can do a null check of GoogleDefaultCredentials, to catch some auth errors early.